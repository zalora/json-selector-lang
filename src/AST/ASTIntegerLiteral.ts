// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.

import ASTExpression from './ASTExpression';
import { Token } from '../Lexer';

class ASTIntegerLiteral implements ASTExpression {
  token: Token;
  value: string;

  constructor(token: Token, value: string) {
    this.token = token;
    this.value = value;
  }

  getDescription(): string {
    return this.value || '';
  }

  tokenLiteral(): string {
    const { literal = '' } = this.token || {};

    return literal;
  }
}

export default ASTIntegerLiteral;
