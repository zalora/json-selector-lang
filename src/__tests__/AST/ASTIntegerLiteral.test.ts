// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.

import ASTIntegerLiteral from '../../AST/ASTIntegerLiteral';
import { int } from '../../tokens';

describe('ASTIntegerLiteral', () => {
  const token = { type: int, literal: '2' };

  it('tests instance with token input', () => {
    const astInt = new ASTIntegerLiteral(token, parseInt(token.literal));
    expect(astInt).toMatchSnapshot();
  });

  it('tests toNumber returns value of the identifier', () => {
    const astInt = new ASTIntegerLiteral(token, parseInt(token.literal));
    expect(astInt.toNumber()).toMatchSnapshot();
  });

  it('tests tokenLiteral returns literal of the token', () => {
    const astInt = new ASTIntegerLiteral(token, parseInt(token.literal));
    expect(astInt.tokenLiteral()).toMatchSnapshot();
  });
});
