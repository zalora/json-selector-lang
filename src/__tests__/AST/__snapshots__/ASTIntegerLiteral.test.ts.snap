// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASTIntegerLiteral tests instance with token input 1`] = `
ASTIntegerLiteral {
  "token": Object {
    "literal": "2",
    "type": Symbol(int),
  },
  "value": 2,
}
`;

exports[`ASTIntegerLiteral tests toNumber returns value of the identifier 1`] = `2`;

exports[`ASTIntegerLiteral tests tokenLiteral returns literal of the token 1`] = `"2"`;
